redbot/core/audio_utils/lavalink.py

```diff
@@ -0,0 +1,234 @@
+                                    "Attempting to connect to an existing server...")
+                    except LavalinkDownloadFailed as exc:
+                        await asyncio.sleep(1)
+                        if exc.should_retry:
+                            log.exception(
+                                "Exception whilst starting internal Lavalink server, retrying...",
+                                exc_info=exc,
+                            )
+                            continue
+                        else:
+                            log.exception(
+                                "Fatal exception whilst starting internal Lavalink server, "
+                                "aborting...",
+                                exc_info=exc,
+                            )
+                            self._lavalink_connection_aborted = True
+                            raise
+                    except asyncio.CancelledError:
+                        log.exception("Invalid machine architecture, cannot run Lavalink.")
+                        raise
+                    except Exception as exc:
+                        log.exception(
+                            "Unhandled exception whilst starting internal Lavalink server, "
+                            "aborting...",
+                            exc_info=exc,
+                        )
+                        self._lavalink_connection_aborted = True
+                        raise
+                    else:
+                        break
+                else:
+                    break
+            else:
+                host = configs["host"]
+                password = configs["password"]
+                ws_port = configs["ws_port"]
+                break
+        else:
+            log.critical(
+                "Setting up the Lavalink server failed after multiple attempts. "
+                "See above tracebacks for details."
+            )
+            self._lavalink_connection_aborted = True
+            return
+
+        async for _ in AsyncIter(range(0, max_retries)):
+            if lavalink.node._nodes:
+                await lavalink.node.disconnect()
+            try:
+                await lavalink.initialize(
+                    bot=self._bot,
+                    host=host,
+                    password=password,
+                    ws_port=ws_port,
+                    timeout=timeout,
+                    resume_key=f"Red-Core-Audio-{self._bot.user.id}-{data_manager.instance_name}"
+                )
+                if IS_DEBUG:
+                    log.debug("Lavalink connection established")
+            except asyncio.TimeoutError:
+                log.error("Connecting to Lavalink server timed out, retrying...")
+                if not external and self._server_manager.is_running:
+                    await self._server_manager.shutdown_ll_server()
+                await asyncio.sleep()
```

Comment:
Hey Benno, just noticed this when external Lavalink is configured but unavailable:
![traceback](https://img.vexcodes.com/IYV8xE)